src/compile/unit.ts

```diff
@@ -244,16 +244,20 @@ export class UnitModel extends Model {
     }, {});
   }
 
-  private initLegend(encoding: Encoding, config: Config): Dict<Legend> {
+  private initLegend(encoding: Encoding): Dict<Legend> {
     return NONSPATIAL_SCALE_CHANNELS.reduce(function(_legend, channel) {
       const channelDef = encoding[channel];
       if (isFieldDef(channelDef)) {
         const legendSpec = channelDef.legend;
         // We no longer support false in the schema, but we keep false here for backward compatability.
+        let vlLegendBase: VlLegendBase = {};
```

Comment:
1) this logic is definitely wrong.  Why copy `encoding.<legend-base-property>`, which doesn't exists?

2) Bad code tab alignment.